compiler/rustc_hir_typeck/src/expr.rs

```diff
@@ -3412,10 +3423,21 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     if let Ok(index) = field.as_str().parse::<usize>()
                         && field.name == sym::integer(index)
                     {
-                        for ty in tys.iter().take(index + 1) {
-                            self.require_type_is_sized(ty, expr.span, ObligationCauseCode::Misc);
-                        }
```

Comment:
What's the justification for removing this? Trivial that this is implied by the well-formedness of the type?

For annoying level of completetness, is there a test that exercises this? If not, pls add. Maybe something like `offset_of!(([i32], i32), 1)`

Reply:
I made sure all tests still pass and couldn't see any reason why the loop existed, and there was no comment explaining it, so I removed it.

We don't have a similar loop for ADTs either. Not sure what is the point of a test for that...